core/src/main/java/org/hibernate/ogm/cfg/impl/HibernateSearchIntegration.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.cfg.impl;
+
+import java.util.Map;
+import java.util.Properties;
+
+import org.hibernate.cfg.Configuration;
+
+/**
+ * Hibernate Search is an optional dependency, so while we need refer to its constants
+ * to make sure its configured correctly for OGM by default, we don't want to create a
+ * hard dependency. This class should be used as a helper to centralise that.
+ *
+ * @author Sanne Grinovero
+ */
+public final class HibernateSearchIntegration {
+
+	private static final boolean searchIsAvailable = isSearchAvailable();
+	private static String MASSINDEXER_PROPERTY_KEY = null;
```

Comment:
ah I misunderstood that, I had tried in a different way.
Happy to return on this and polish, but why would you not extract constants? they are repeated all over.